src/client/shader.cpp

```diff
@@ -140,8 +141,25 @@ class SourceShaderCache
 			type = "geometry";
 		else if (str_ends_with(filename, ".fs.glsl"))
 			type = "fragment";
-		else
+		else {
 			errorstream << "Unsupported shader type: " << filename << std::endl;
+			return;
+		}
+		if (program.length() > 10000) {
+			errorstream << "Rejecting long shader (> 10k chars): " << filename << std::endl;
+			return;
+		}
+
+		const std::regex line_comments("//.*?\n");
+		const std::string no_line_comments = std::regex_replace(program, line_comments, "");
+		const std::regex multiline_comments("/\\*.*?\\*\n");
+		const std::string no_comments = std::regex_replace(no_line_comments, multiline_comments, "");
+		const std::regex long_identifier_or_keyword("[a-zA-Z][a-zA-Z0-9]{32,}");
+		std::smatch match;
+		if (std::regex_search(no_comments, match, long_identifier_or_keyword)) {
+			errorstream << "Rejecting shader with long identifier (> 32 chars): " << match.str(0) << " in " << filename << std::endl;
```

Comment:
What’s the point?

Reply:
> The shader section states clearly that shaders themselves are not the problem, you just need to sanitize identifiers to prevent tripping potential bugs in the compiler. This can be as simple as rejecting or mangling alphanumeric blobs longer than a certain length.
\- hecktest